sign_up.php

```diff
@@ -44,68 +35,40 @@
     }
 
     if (empty($error[$pic_name]) && $fields_validated) {
-
-        $sql = "INSERT INTO `users` ( `first_name`, `middle_name`, `last_name`, `gender`, `dob`,"
-                . " `type`, `bio`, `preferred_comm`, `mobile`, `created_date`) VALUES ('"
-                . $_POST['first_name'] . "','". $_POST['middle_name'] . "','". $_POST['last_name'] 
-                . "','". $_POST['gender'] . "','". $_POST['dob'] . "','". $_POST['user_type'] 
-                . "','". $_POST['comment'] . "','". implode(',', $_POST['pref_comm']) . "','"
-                . $_POST['contact_num'] . "', NOW())";
-
-        if ( ! mysqli_query($conn, $sql)) {
-            $msg = "New record in USERS FAILURE";
-            echo 'err1-->' . mysqli_error($conn);
-            header('Location: error.php');
-        }
-
-        $user_id = mysqli_insert_id($conn);
-        $sql_login = "INSERT INTO `login`(`email`, `password`, `user_id`, `created_date`) VALUES ('"
-                . $_POST['email']. "','". $_POST['password']. "','". $user_id. "', NOW())";
-
-        if ( ! mysqli_query($conn, $sql_login)) {
-            $msg = "New record in LOGIN FAILURE";
-            echo 'err2-->' . mysqli_error($conn);
-        }
-
-        $sql_addr = "INSERT INTO `user_address` (`user_id`, `type`, `street`, `city`, `state`, `zip`"
-                . ",`created_date`) VALUES ('". $user_id. "', '1', '" . $_POST['res_addrstreet']
-                . "', '". $_POST['res_addrcity']. "', '". $_POST['res_addrstate']. "', '". $_POST['res_addrzip']
-                . "', NOW())";
-
+        
+        $data =['first_name'=> $_POST['first_name'], 'middle_name'=> $_POST['middle_name'],
+            'last_name'=> $_POST['last_name'], 'gender'=> $_POST['gender'], 'dob'=> $_POST['dob'],
+            'type'=> $_POST['user_type'],'bio'=> $_POST['comment'],'preferred_comm'=> implode(',', $_POST['pref_comm']),
+            'mobile'=> $_POST['contact_num']];
+        $user_id = $db->insert_or_update(1, 'users', $data);
+        
+        $data_login = ['email'=> $_POST['email'],'password'=> $_POST['password'], 'user_id'=> $user_id];
+        $db->insert_or_update(1, 'login', $data_login);
+        
+        $data_res_addr = ['user_id'=> $user_id, 'type'=> '1', 'street'=> $_POST['res_addrstreet'],
+            'city'=> $_POST['res_addrcity'], 'state'=> $_POST['res_addrstate'], 'zip'=> $_POST['res_addrzip']]; 
+        $db->insert_or_update(1, 'user_address', $data_res_addr);
+        
```

Comment:
In some paces you are using space before and after = in assignment. And somewhere not.
Make consistency

$a = 1;
or 
$a=1;